### ISSUE TITLE
RequestDelegateGenerator creates incorrect code with default parameters for bool and CancellationToken in Lambda expressions (AOT)

### ISSUE THREAD

**Issue by christiannagel** [dotnet/aspnetcore#51183]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Creating an AOT Web API project (dotnet new webapiaot), and using default parameter values in Lambda expression for bool and CancellationToken, such as the following creates invalid code.

```csharp
        group.MapGet("/", async (
            IGamesService gameService,
            string? gameType = default,
            string? playerName = default,
            DateOnly? date = default,
            bool ended = default,
            CancellationToken cancellationToken = default) =>
        {
          //...
        });
```

This is part of the generated code:

```csharp
                var handler = Cast(del, global::System.Threading.Tasks.Task<global::Microsoft.AspNetCore.Http.HttpResults.Ok<global::System.Collections.Generic.IEnumerable<global::Codebreaker.GameAPIs.Models.Game>>> (global::Codebreaker.GameAPIs.Services.IGamesService arg0, global::System.String? arg1= null, global::System.String? arg2= null, global::System.DateOnly? arg3= null, global::System.Boolean arg4= False, global::System.Threading.CancellationToken arg5= null) => throw null!);
```

With the `bool` parameter, the value `False` is created which should be `false`.
With the default value for the `CancellationToken`, `null` is assigned - but this is a vlaue type.

### Expected Behavior

I would expect to create true/false for bool types, and correct defaults for value types such as CancellationToken.

### Steps To Reproduce

Create an AOT Web API project (webapiaot template)
Define an API method with bool and CancellationToken parameters, and assign default values

### .NET Version

8.0.100-rc.1.23455.8

### Anything else?

In this repository
https://github.com/PacktPublishing/Pragmatic-Microservices-with-CSharp-and-Azure

in the branch requestdelegategenerator
the issue is in the folder ch05, Codebreaker.GameAPIs.NativeAOT.

**Comment by captainsafia** [comment]
Thanks for the bug report, @christiannagel! I've opened a PR to resolve this issue and will hopefully be able to backport it to 8.0 RTM.

**Comment by captainsafia** [comment]
> Thanks for the bug report, @christiannagel! I've opened a PR to resolve this issue and will hopefully be able to backport it to 8.0 RTM.

Missed the deadline for 8.0 RTM but planning on taking this in for the first servicing release of 8.0.

@christiannagel Does that timeline work for the updates you're making to your content?

**Comment by christiannagel** [comment]
@captainsafia - yes, this works, thanks for asking! My new book is planned to be released mid-2024 :-)

**Comment by christiannagel** [comment]
fixed with 8.0.2 - thanks!
